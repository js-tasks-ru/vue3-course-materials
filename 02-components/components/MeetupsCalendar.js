import { defineComponent } from '../vendor/vue.esm-browser.js';

export default defineComponent({
  name: 'MeetupsCalendar',

  props: {
    meetups: {
      type: Array,
      required: true,
    },
  },

  template: `<div class="meetups-calendar">Календарь</div>`,
});
