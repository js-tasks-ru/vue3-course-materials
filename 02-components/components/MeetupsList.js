import { defineComponent } from '../vendor/vue.esm-browser.js';
import MeetupCard from './MeetupCard.js';

export default defineComponent({
  name: 'MeetupsList',

  components: {
    MeetupCard,
  },

  props: {
    meetups: {
      type: Array,
      required: true,
    },
  },

  template: `
    <ul class="meetups-list">
      <li v-for="meetup in meetups" :key="meetup.id" class="meetups-list__item">
        <a :href="\`/meetups/\${meetup.id}\`" class="meetups-list__item-link" tabindex="0">
          <MeetupCard :meetup="meetup" />
        </a>
      </li>
    </ul>
  `,
});
