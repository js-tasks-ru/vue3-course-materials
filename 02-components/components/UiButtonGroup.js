import { defineComponent } from '../vendor/vue.esm-browser.js';

export default defineComponent({
  name: 'UiButtonGroup',

  props: {
    view: {
      type: String,
      default: 'list',
      validator: (value) => ['list', 'calendar'].includes(value),
    },
  },

  emits: {
    'update:view': (value) => ['list', 'calendar'].includes(value),
  },

  methods: {
    select(value) {
      this.$emit('update:view', value);
    },
  },

  template: `
    <div class="button-group" role="group">
      <button
        type="button"
        class="button-group__button"
        :class="{ 'button-group__button_active': view === 'list' }"
        aria-label="Meetups list"
        :aria-selected="view === 'list'"
        @click="select('list')"
      >
        <svg fill="none" height="28" viewBox="0 0 28 28" width="28" xmlns="http://www.w3.org/2000/svg">
          <path
            clip-rule="evenodd"
            d="m5 7c-.55228 0-1 .44772-1 1s.44772 1 1 1h.01c.55228 0 1-.44772 1-1s-.44772-1-1-1zm5 0c-.55228 0-1 .44772-1 1s.44772 1 1 1h13c.5523 0 1-.44772 1-1s-.4477-1-1-1zm0 6c-.55228 0-1 .4477-1 1s.44772 1 1 1h13c.5523 0 1-.4477 1-1s-.4477-1-1-1zm-1 7c0-.5523.44772-1 1-1h13c.5523 0 1 .4477 1 1s-.4477 1-1 1h-13c-.55228 0-1-.4477-1-1zm-5-6c0-.5523.44772-1 1-1h.01c.55228 0 1 .4477 1 1s-.44772 1-1 1h-.01c-.55228 0-1-.4477-1-1zm1 5c-.55228 0-1 .4477-1 1s.44772 1 1 1h.01c.55228 0 1-.4477 1-1s-.44772-1-1-1z"
            fill-rule="evenodd"
          ></path>
        </svg>
      </button>
      <button
        type="button"
        class="button-group__button"
        :class="{ 'button-group__button_active': view === 'calendar' }"
        aria-label="Meetups calendar"
        :aria-selected="view === 'calendar'"
        @click="select('calendar')"
      >
        <svg height="22" viewBox="0 0 20 22" width="20" xmlns="http://www.w3.org/2000/svg">
          <path
            clip-rule="evenodd"
            d="m15 1c0-.552285-.4477-1-1-1s-1 .447715-1 1v1h-6v-1c0-.552285-.44772-1-1-1s-1 .447715-1 1v1h-2c-1.65685 0-3 1.34315-3 3v14c0 1.6569 1.34315 3 3 3h14c1.6569 0 3-1.3431 3-3v-14c0-1.65685-1.3431-3-3-3h-2zm3 7v-3c0-.55228-.4477-1-1-1h-2v1c0 .55228-.4477 1-1 1s-1-.44772-1-1v-1h-6v1c0 .55228-.44772 1-1 1s-1-.44772-1-1v-1h-2c-.55228 0-1 .44772-1 1v3zm-16 2h16v9c0 .5523-.4477 1-1 1h-14c-.55228 0-1-.4477-1-1z"
            fill-rule="evenodd"
          ></path>
        </svg>
      </button>
    </div>
  `,
});
