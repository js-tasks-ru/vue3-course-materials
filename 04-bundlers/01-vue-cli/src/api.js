/** URL адрес API */
export const API_URL = process.env.VUE_APP_API_URL;

/**
 * Получает список митапов с API
 * @return {Promise}
 */
export async function fetchMeetups() {
  return fetch(`${API_URL}/meetups`).then((res) => res.json());
}
