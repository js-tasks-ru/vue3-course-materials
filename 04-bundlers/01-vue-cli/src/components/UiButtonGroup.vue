<template>
  <div class="button-group" role="group">
    <button
      type="button"
      class="button-group__button"
      :class="{ 'button-group__button_active': view === 'list' }"
      aria-label="Список"
      :aria-selected="view === 'list'"
      @click="select('list')"
    >
      <svg fill="none" height="28" viewBox="0 0 28 28" width="28" xmlns="http://www.w3.org/2000/svg">
        <path
          clip-rule="evenodd"
          d="m5 7c-.55228 0-1 .44772-1 1s.44772 1 1 1h.01c.55228 0 1-.44772 1-1s-.44772-1-1-1zm5 0c-.55228 0-1 .44772-1 1s.44772 1 1 1h13c.5523 0 1-.44772 1-1s-.4477-1-1-1zm0 6c-.55228 0-1 .4477-1 1s.44772 1 1 1h13c.5523 0 1-.4477 1-1s-.4477-1-1-1zm-1 7c0-.5523.44772-1 1-1h13c.5523 0 1 .4477 1 1s-.4477 1-1 1h-13c-.55228 0-1-.4477-1-1zm-5-6c0-.5523.44772-1 1-1h.01c.55228 0 1 .4477 1 1s-.44772 1-1 1h-.01c-.55228 0-1-.4477-1-1zm1 5c-.55228 0-1 .4477-1 1s.44772 1 1 1h.01c.55228 0 1-.4477 1-1s-.44772-1-1-1z"
          fill-rule="evenodd"
        ></path>
      </svg>
    </button>
    <button
      type="button"
      class="button-group__button"
      :class="{ 'button-group__button_active': view === 'calendar' }"
      aria-label="Календарь"
      :aria-selected="view === 'calendar'"
      @click="select('calendar')"
    >
      <svg height="22" viewBox="0 0 20 22" width="20" xmlns="http://www.w3.org/2000/svg">
        <path
          clip-rule="evenodd"
          d="m15 1c0-.552285-.4477-1-1-1s-1 .447715-1 1v1h-6v-1c0-.552285-.44772-1-1-1s-1 .447715-1 1v1h-2c-1.65685 0-3 1.34315-3 3v14c0 1.6569 1.34315 3 3 3h14c1.6569 0 3-1.3431 3-3v-14c0-1.65685-1.3431-3-3-3h-2zm3 7v-3c0-.55228-.4477-1-1-1h-2v1c0 .55228-.4477 1-1 1s-1-.44772-1-1v-1h-6v1c0 .55228-.44772 1-1 1s-1-.44772-1-1v-1h-2c-.55228 0-1 .44772-1 1v3zm-16 2h16v9c0 .5523-.4477 1-1 1h-14c-.55228 0-1-.4477-1-1z"
          fill-rule="evenodd"
        ></path>
      </svg>
    </button>
  </div>
</template>

<script>
export default {
  name: 'UiButtonGroup',

  props: {
    view: {
      type: String,
      default: 'list',
      validator: (value) => ['list', 'calendar'].includes(value),
    },
  },

  emits: ['update:view'],

  methods: {
    select(value) {
      this.$emit('update:view', value);
    },
  },
};
</script>

<style scoped>
.button-group {
  display: inline-flex;
  flex-direction: row;
  flex-wrap: wrap;
  white-space: nowrap;
  width: 96px;
}

.button-group__button {
  height: 44px;
  width: 48px;
  border: 2px solid var(--blue);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  transition-duration: 0.3s;
  transition-property: background-color, fill;
  background-color: var(--white);
  box-shadow: none;
  outline: none;
  cursor: pointer;
}

.button-group__button:first-child {
  border-radius: 22px 0 0 22px;
  padding-left: 2px;
}

.button-group__button:last-child {
  border-radius: 0 22px 22px 0;
  padding-right: 2px;
}

.button-group__button.button-group__button_active {
  background-color: var(--blue);
}

.button-group__button svg,
.button-group__button svg {
  fill: var(--blue);
}

.button-group__button:hover,
.button-group__button.button-group__button_active {
  background-color: var(--blue);
}

.button-group__button:hover svg,
.button-group__button.button-group__button_active svg {
  fill: var(--white);
}
</style>
