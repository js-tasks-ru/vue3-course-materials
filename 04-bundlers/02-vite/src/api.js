/** URL адрес API */
export const API_URL = import.meta.env.VITE_API_URL;

/**
 * Получает список митапов с API
 * @return {Promise}
 */
export async function fetchMeetups() {
  return fetch(`${API_URL}/meetups`).then((res) => res.json());
}
