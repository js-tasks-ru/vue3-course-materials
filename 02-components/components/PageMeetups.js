import { defineComponent } from '../vendor/vue.esm-browser.js';
import UiRadioGroup from './UiRadioGroup.js';
import MeetupsList from './MeetupsList.js';
import UiContainer from './UiContainer.js';
import MeetupsCalendar from './MeetupsCalendar.js';
import UiAlert from './UiAlert.js';
import UiButtonGroup from './UiButtonGroup.js';

const fetchMeetups = () => fetch('./api/meetups.json').then((res) => res.json());

export default defineComponent({
  name: 'PageMeetups',

  dateFilterOptions: [
    { text: 'Все', value: 'all' },
    { text: 'Прошедшие', value: 'past' },
    { text: 'Ожидаемые', value: 'future' },
  ],

  components: {
    MeetupsList,
    MeetupsCalendar,
    UiRadioGroup,
    UiButtonGroup,
    UiContainer,
    UiAlert,
  },

  data() {
    return {
      meetups: null,

      filter: {
        date: 'all',
        participation: 'all',
        search: '',
      },

      view: 'list',
    };
  },

  computed: {
    filteredMeetups() {
      const dateFilter = (meetup) =>
        this.filter.date === 'all' ||
        (this.filter.date === 'past' && new Date(meetup.date) <= new Date()) ||
        (this.filter.date === 'future' && new Date(meetup.date) > new Date());

      const participationFilter = (meetup) =>
        this.filter.participation === 'all' ||
        (this.filter.participation === 'organizing' && meetup.organizing) ||
        (this.filter.participation === 'attending' && meetup.attending);

      const searchFilter = (meetup) =>
        [meetup.title, meetup.description, meetup.place, meetup.organizer]
          .join(' ')
          .toLowerCase()
          .includes(this.filter.search.toLowerCase());

      return this.meetups.filter((meetup) => dateFilter(meetup) && participationFilter(meetup) && searchFilter(meetup));
    },
  },

  async mounted() {
    this.meetups = await fetchMeetups();
  },

  template: `
    <UiContainer>
      <div class="filters-panel">
        <div class="filters-panel__col">
          <UiRadioGroup v-model="filter.date" :options="$options.dateFilterOptions"  />
        </div>

        <div class="filters-panel__col">
          <div class="form-group form-group_inline">
            <div class="input-group input-group_icon input-group_icon-left">
              <div class="input-group__icon">
                <img class="icon" src="/assets/icons/icon-search.svg" alt="search" />
              </div>

              <input
                class="form-control form-control_rounded form-control_sm"
                placeholder="Поиск"
                type="search"
                v-model.trim="filter.search"
              />
            </div>
          </div>
          <div class="form-group form-group_inline">
            <UiButtonGroup v-model:view="view" />
          </div>
        </div>
      </div>

      <template v-if="meetups">
        <template v-if="filteredMeetups.length">
          <MeetupsList v-if="view === 'list'" :meetups="filteredMeetups" />
          <MeetupsCalendar v-else-if="view === 'calendar'" :meetups="filteredMeetups" />
        </template>
        <UiAlert v-else >Митапов по заданным условиям не найдено...</UiAlert>
      </template>
      <UiAlert v-else text="Загрузка..." />
    </UiContainer>
  `,
});
