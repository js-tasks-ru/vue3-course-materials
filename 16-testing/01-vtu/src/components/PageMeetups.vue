<template>
  <UiContainer>
    <div class="filters-panel">
      <div class="filters-panel__col">
        <UiRadioGroup v-model="filter.date" :options="$options.dateFilterOptions" name="date" />
      </div>

      <div class="filters-panel__col">
        <div class="form-group form-group_inline">
          <UiInput v-model.trim="filter.search" rounded small placeholder="Поиск" type="search">
            <template #left-icon>
              <ui-icon icon="search" />
            </template>
          </UiInput>
        </div>
        <div class="form-group form-group_inline">
          <UiButtonGroup v-model:view="view" />
        </div>
      </div>
    </div>

    <KeepAlive v-if="meetups" include="MeetupsCalendar">
      <component :is="viewComponent" v-if="filteredMeetups.length" :meetups="filteredMeetups" />
      <UiAlert v-else>Митапов по заданным условиям не найдено...</UiAlert>
    </KeepAlive>
    <UiAlert v-else>Загрузка...</UiAlert>
  </UiContainer>
</template>

<script>
import MeetupsList from './MeetupsList.vue';
import MeetupsCalendar from './MeetupsCalendar.vue';
import UiRadioGroup from './UiRadioGroup.vue';
import UiButtonGroup from './UiButtonGroup.vue';
import UiContainer from './UiContainer.vue';
import UiAlert from './UiAlert.vue';
import UiIcon from './UiIcon.vue';
import { fetchMeetups } from '../api';
import UiInput from './UiInput.vue';

export default {
  name: 'PageMeetups',

  dateFilterOptions: [
    { text: 'Все', value: 'all' },
    { text: 'Прошедшие', value: 'past' },
    { text: 'Ожидаемые', value: 'future' },
  ],

  components: {
    UiInput,
    UiIcon,
    UiRadioGroup,
    UiButtonGroup,
    UiContainer,
    UiAlert,
  },

  data() {
    return {
      meetups: null,

      filter: {
        date: 'all',
        participation: 'all',
        search: '',
      },

      view: 'list',
    };
  },

  computed: {
    filteredMeetups() {
      if (!this.meetups) {
        return null;
      }

      const dateFilter = (meetup) =>
        this.filter.date === 'all' ||
        (this.filter.date === 'past' && new Date(meetup.date) <= new Date()) ||
        (this.filter.date === 'future' && new Date(meetup.date) > new Date());

      const participationFilter = (meetup) =>
        this.filter.participation === 'all' ||
        (this.filter.participation === 'organizing' && meetup.organizing) ||
        (this.filter.participation === 'attending' && meetup.attending);

      const searchFilter = (meetup) =>
        [meetup.title, meetup.description, meetup.place, meetup.organizer]
          .join(' ')
          .toLowerCase()
          .includes(this.filter.search.toLowerCase());

      return this.meetups.filter((meetup) => dateFilter(meetup) && participationFilter(meetup) && searchFilter(meetup));
    },

    viewComponent() {
      const viewToComponents = {
        list: MeetupsList,
        calendar: MeetupsCalendar,
      };
      return viewToComponents[this.view];
    },
  },

  mounted() {
    fetchMeetups().then((meetups) => {
      this.meetups = meetups;
    });
  },
};
</script>

<style scoped>
.filters-panel {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 40px 0 32px 0;
}

.filters-panel__col {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
}

@media all and (max-width: 767px) {
  .filters-panel {
    flex-direction: column;
  }

  .filters-panel__col + .filters-panel__col {
    margin-top: 16px;
  }
}
</style>
