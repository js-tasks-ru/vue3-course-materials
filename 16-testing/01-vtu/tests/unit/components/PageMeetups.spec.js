import { shallowMount, mount, flushPromises } from '@vue/test-utils';
import { nextTick } from 'vue';
import PageMeetups from '@/components/PageMeetups';
import MeetupsList from '@/components/MeetupsList';
import { fetchMeetups } from '@/api';
import meetups from '../__fixtures__/meetups.json';
import UiButtonGroup from '@/components/UiButtonGroup';
import MeetupsCalendar from '@/components/MeetupsCalendar';

jest.mock('@/api');

describe('PageMeetups', () => {
  beforeEach(() => {
    jest.resetModules();
    fetchMeetups.mockResolvedValue(meetups);
  });

  it('should be defined', () => {
    expect(PageMeetups).toBeDefined();
  });

  describe('Unit testing', () => {
    it('should fetch meetups by fetchMeetups on mounted', async () => {
      const wrapper = mount(PageMeetups, {
        global: {
          stubs: {
            UiIcon: true,
            UiEmpty: true,
            MeetupsCalendar: true,
            UiButtonGroup: true,
            UiRadioGroup: true,
            UiInput: true,
            MeetupsList: true,
            KeepAlive: true,
          },
        },
      });
      await flushPromises();
      expect(fetchMeetups).toHaveBeenCalled();
    });

    it('should render meetups by MeetupsList', async () => {
      const wrapper = mount(PageMeetups, {
        global: {
          stubs: {
            UiButtonGroup: true,
            UiContainer: true,
            UiEmpty: true,
            UiIcon: true,
            UiInput: true,
            UiRadioGroup: true,
            MeetupsCalendar: true,
            MeetupsList: true,
            KeepAlive: true,
          },
        },
      });
      await flushPromises();
      expect(fetchMeetups).toHaveBeenCalled();
      expect(wrapper.getComponent(MeetupsList).props('meetups')).toMatchObject(meetups);
    });

    it('should render meetups by MeetupsList (with shallowMount)', async () => {
      const wrapper = shallowMount(PageMeetups);
      await flushPromises();
      expect(fetchMeetups).toHaveBeenCalled();
      expect(wrapper.getComponent(MeetupsList).props('meetups')).toMatchObject(meetups);
    });

    it('should render MeetupsCalendar instead of MeetupsList after select calendar by UiButtonGroup', async () => {
      const wrapper = shallowMount(PageMeetups);
      await flushPromises();
      const buttonGroup = wrapper.getComponent(UiButtonGroup);
      buttonGroup.vm.$emit('update:view', 'calendar');
      await nextTick();
      expect(wrapper.findComponent(MeetupsList).exists()).toBeFalsy();
      expect(wrapper.findComponent(MeetupsCalendar).exists()).toBeTruthy();
    });

    it('should match shallowMount snapshot', async () => {
      const wrapper = shallowMount(PageMeetups);
      await flushPromises();
      expect(wrapper.__app._container).toMatchSnapshot();
    });
  });

  describe('Integration testing', () => {
    it('should render meetups headers in meetup-card header', async () => {
      const wrapper = mount(PageMeetups);
      await flushPromises();
      expect(fetchMeetups).toHaveBeenCalled();
      expect(wrapper.getComponent(MeetupsList).props('meetups')).toMatchObject(meetups);
      // Выглядит не очень хорошо. Лучше снапшот?
      const cardHeaders = wrapper.findAll('.meetup-card header');
      for (let i = 0; i < cardHeaders.length; i++) {
        expect(cardHeaders[i].text()).toContain(meetups[i].title);
      }
    });

    it('should render .meetups-calendar instead of .meetups-list after click on calendar button', async () => {
      const wrapper = mount(PageMeetups);
      await flushPromises();
      // const calendarViewButton = wrapper.getAll('.button-group__button')[1];
      const calendarViewButton = wrapper.get('[aria-label="Календарь"]');
      await calendarViewButton.trigger('click');
      expect(wrapper.find('.meetups-list').exists()).toBeFalsy();
      expect(wrapper.find('.meetups-calendar').exists()).toBeTruthy();
    });

    it('should match mount snapshot', async () => {
      const wrapper = mount(PageMeetups);
      await flushPromises();
      expect(wrapper.__app._container).toMatchSnapshot();
    });
  });
});
