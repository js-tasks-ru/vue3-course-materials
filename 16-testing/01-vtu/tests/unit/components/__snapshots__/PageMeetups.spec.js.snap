// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMeetups Integration testing should match mount snapshot 1`] = `
<div
  data-v-app=""
>
  <div
    class="container"
  >
    
    <div
      class="filters-panel"
    >
      <div
        class="filters-panel__col"
      >
        <div
          class="radio-group"
        >
          
          <div
            class="radio-group__button"
          >
            <input
              class="radio-group__input"
              id="radio-buttons_date_all"
              name="date"
              type="radio"
            />
            <label
              class="radio-group__label"
              for="radio-buttons_date_all"
            >
              Все
            </label>
          </div>
          <div
            class="radio-group__button"
          >
            <input
              class="radio-group__input"
              id="radio-buttons_date_past"
              name="date"
              type="radio"
            />
            <label
              class="radio-group__label"
              for="radio-buttons_date_past"
            >
              Прошедшие
            </label>
          </div>
          <div
            class="radio-group__button"
          >
            <input
              class="radio-group__input"
              id="radio-buttons_date_future"
              name="date"
              type="radio"
            />
            <label
              class="radio-group__label"
              for="radio-buttons_date_future"
            >
              Ожидаемые
            </label>
          </div>
          
        </div>
      </div>
      <div
        class="filters-panel__col"
      >
        <div
          class="form-group form-group_inline"
        >
          <div
            class="input-group input-group_icon input-group_icon-left"
          >
            <div
              class="input-group__icon"
            >
              
              <img
                alt="search"
                class="icon"
                src=""
              />
              
            </div>
            <input
              class="form-control form-control_rounded form-control_sm"
              modelmodifiers="[object Object]"
              placeholder="Поиск"
              type="search"
            />
            <!-- Или вместо customModel: -->
            <!-- :value="modelValue"-->
            <!-- @input="$emit('update:modelValue', $event.target.value)"-->
          </div>
        </div>
        <div
          class="form-group form-group_inline"
        >
          <div
            class="button-group"
            role="group"
          >
            <button
              aria-label="Список"
              aria-selected="true"
              class="button-group__button button-group__button_active"
              type="button"
            >
              <svg
                fill="none"
                height="28"
                viewBox="0 0 28 28"
                width="28"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="m5 7c-.55228 0-1 .44772-1 1s.44772 1 1 1h.01c.55228 0 1-.44772 1-1s-.44772-1-1-1zm5 0c-.55228 0-1 .44772-1 1s.44772 1 1 1h13c.5523 0 1-.44772 1-1s-.4477-1-1-1zm0 6c-.55228 0-1 .4477-1 1s.44772 1 1 1h13c.5523 0 1-.4477 1-1s-.4477-1-1-1zm-1 7c0-.5523.44772-1 1-1h13c.5523 0 1 .4477 1 1s-.4477 1-1 1h-13c-.55228 0-1-.4477-1-1zm-5-6c0-.5523.44772-1 1-1h.01c.55228 0 1 .4477 1 1s-.44772 1-1 1h-.01c-.55228 0-1-.4477-1-1zm1 5c-.55228 0-1 .4477-1 1s.44772 1 1 1h.01c.55228 0 1-.4477 1-1s-.44772-1-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <button
              aria-label="Календарь"
              aria-selected="false"
              class="button-group__button"
              type="button"
            >
              <svg
                height="22"
                viewBox="0 0 20 22"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="m15 1c0-.552285-.4477-1-1-1s-1 .447715-1 1v1h-6v-1c0-.552285-.44772-1-1-1s-1 .447715-1 1v1h-2c-1.65685 0-3 1.34315-3 3v14c0 1.6569 1.34315 3 3 3h14c1.6569 0 3-1.3431 3-3v-14c0-1.65685-1.3431-3-3-3h-2zm3 7v-3c0-.55228-.4477-1-1-1h-2v1c0 .55228-.4477 1-1 1s-1-.44772-1-1v-1h-6v1c0 .55228-.44772 1-1 1s-1-.44772-1-1v-1h-2c-.55228 0-1 .44772-1 1v3zm-16 2h16v9c0 .5523-.4477 1-1 1h-14c-.55228 0-1-.4477-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <ul
      class="meetups-list"
    >
      
      <li
        class="meetups-list__item"
      >
        <a
          class="meetups-list__item-link"
          href="/meetups/2"
          tabindex="0"
        >
          <article
            class="card meetup-card"
          >
            <div
              class="card__col"
            >
              <div
                class="card__cover"
              >
                <header>
                  
                  Демо-Митап
                  
                </header>
              </div>
            </div>
            <div
              class="card__col"
            >
              <div
                class="card__content"
              >
                
                <span
                  class="badge badge_success meetup-card__badge"
                >
                  
                  Организую
                  
                </span>
                <!--v-if-->
                <ul
                  class="meetup-info"
                >
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="user"
                      class="icon meetup-info__icon"
                      src=""
                    />
                     Demo Organizer
                  </li>
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="map"
                      class="icon meetup-info__icon"
                      src=""
                    />
                     Internet
                  </li>
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="cal-lg"
                      class="icon meetup-info__icon"
                      src=""
                    />
                    <time
                      datetime="2020-06-18"
                    >
                      June 18, 2020
                    </time>
                  </li>
                </ul>
                
              </div>
            </div>
          </article>
        </a>
      </li>
      <li
        class="meetups-list__item"
      >
        <a
          class="meetups-list__item-link"
          href="/meetups/5"
          tabindex="0"
        >
          <article
            class="card meetup-card"
          >
            <div
              class="card__col"
            >
              <div
                class="card__cover"
              >
                <header>
                  
                  Vue.js Moscow #3 - Ozon
                  
                </header>
              </div>
            </div>
            <div
              class="card__col"
            >
              <div
                class="card__content"
              >
                
                <!--v-if-->
                <!--v-if-->
                <ul
                  class="meetup-info"
                >
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="user"
                      class="icon meetup-info__icon"
                      src=""
                    />
                     Eugeny F.
                  </li>
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="map"
                      class="icon meetup-info__icon"
                      src=""
                    />
                     Москва, Офис компании Mail.Ru Group
                  </li>
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="cal-lg"
                      class="icon meetup-info__icon"
                      src=""
                    />
                    <time
                      datetime="2020-05-12"
                    >
                      May 12, 2020
                    </time>
                  </li>
                </ul>
                
              </div>
            </div>
          </article>
        </a>
      </li>
      <li
        class="meetups-list__item"
      >
        <a
          class="meetups-list__item-link"
          href="/meetups/6"
          tabindex="0"
        >
          <article
            class="card meetup-card"
          >
            <div
              class="card__col"
            >
              <div
                class="card__cover"
              >
                <header>
                  
                  VueConf US
                  
                </header>
              </div>
            </div>
            <div
              class="card__col"
            >
              <div
                class="card__content"
              >
                
                <!--v-if-->
                <!--v-if-->
                <ul
                  class="meetup-info"
                >
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="user"
                      class="icon meetup-info__icon"
                      src=""
                    />
                     Evan You
                  </li>
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="map"
                      class="icon meetup-info__icon"
                      src=""
                    />
                     USA, AUSTIN CONVENTION CENTER
                  </li>
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="cal-lg"
                      class="icon meetup-info__icon"
                      src=""
                    />
                    <time
                      datetime="2020-05-12"
                    >
                      May 12, 2020
                    </time>
                  </li>
                </ul>
                
              </div>
            </div>
          </article>
        </a>
      </li>
      <li
        class="meetups-list__item"
      >
        <a
          class="meetups-list__item-link"
          href="/meetups/1"
          tabindex="0"
        >
          <article
            class="card meetup-card"
          >
            <div
              class="card__col"
            >
              <div
                class="card__cover"
              >
                <header>
                  
                  MSK VUE.JS MEETUP #1
                  
                </header>
              </div>
            </div>
            <div
              class="card__col"
            >
              <div
                class="card__content"
              >
                
                <!--v-if-->
                <span
                  class="badge badge_primary meetup-card__badge"
                >
                  
                  Участвую
                  
                </span>
                <ul
                  class="meetup-info"
                >
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="user"
                      class="icon meetup-info__icon"
                      src=""
                    />
                     Игорь Ш.
                  </li>
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="map"
                      class="icon meetup-info__icon"
                      src=""
                    />
                     Москва, офис Voximplant (ул. Мытная 66)
                  </li>
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="cal-lg"
                      class="icon meetup-info__icon"
                      src=""
                    />
                    <time
                      datetime="2020-05-08"
                    >
                      May 8, 2020
                    </time>
                  </li>
                </ul>
                
              </div>
            </div>
          </article>
        </a>
      </li>
      <li
        class="meetups-list__item"
      >
        <a
          class="meetups-list__item-link"
          href="/meetups/4"
          tabindex="0"
        >
          <article
            class="card meetup-card"
          >
            <div
              class="card__col"
            >
              <div
                class="card__cover"
              >
                <header>
                  
                  Vue.js Moscow Meetup #2 - Mail.ru
                  
                </header>
              </div>
            </div>
            <div
              class="card__col"
            >
              <div
                class="card__content"
              >
                
                <!--v-if-->
                <!--v-if-->
                <ul
                  class="meetup-info"
                >
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="user"
                      class="icon meetup-info__icon"
                      src=""
                    />
                     Eugeny F.
                  </li>
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="map"
                      class="icon meetup-info__icon"
                      src=""
                    />
                     Москва, Офис компании Mail.Ru Group
                  </li>
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="cal-lg"
                      class="icon meetup-info__icon"
                      src=""
                    />
                    <time
                      datetime="2020-05-05"
                    >
                      May 5, 2020
                    </time>
                  </li>
                </ul>
                
              </div>
            </div>
          </article>
        </a>
      </li>
      <li
        class="meetups-list__item"
      >
        <a
          class="meetups-list__item-link"
          href="/meetups/3"
          tabindex="0"
        >
          <article
            class="card meetup-card"
          >
            <div
              class="card__col"
            >
              <div
                class="card__cover"
              >
                <header>
                  
                  Vue.js Moscow Meetup #1
                  
                </header>
              </div>
            </div>
            <div
              class="card__col"
            >
              <div
                class="card__content"
              >
                
                <!--v-if-->
                <span
                  class="badge badge_primary meetup-card__badge"
                >
                  
                  Участвую
                  
                </span>
                <ul
                  class="meetup-info"
                >
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="user"
                      class="icon meetup-info__icon"
                      src=""
                    />
                     Eugeny F.
                  </li>
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="map"
                      class="icon meetup-info__icon"
                      src=""
                    />
                     Москва, Физтехпарк, офис Acronis
                  </li>
                  <li
                    class="meetup-info__item"
                  >
                    <img
                      alt="cal-lg"
                      class="icon meetup-info__icon"
                      src=""
                    />
                    <time
                      datetime="2020-04-22"
                    >
                      April 22, 2020
                    </time>
                  </li>
                </ul>
                
              </div>
            </div>
          </article>
        </a>
      </li>
      
    </ul>
    
  </div>
</div>
`;

exports[`PageMeetups Unit testing should match shallowMount snapshot 1`] = `
<div
  data-v-app=""
>
  <ui-container-stub>
    <div
      class="filters-panel"
    >
      <div
        class="filters-panel__col"
      >
        <ui-radio-group-stub
          modelvalue="all"
          name="date"
          options="[object Object],[object Object],[object Object]"
        />
      </div>
      <div
        class="filters-panel__col"
      >
        <div
          class="form-group form-group_inline"
        >
          <ui-input-stub
            modelmodifiers="[object Object]"
            modelvalue=""
            placeholder="Поиск"
            rounded="true"
            small="true"
            type="search"
          />
        </div>
        <div
          class="form-group form-group_inline"
        >
          <ui-button-group-stub
            view="list"
          />
        </div>
      </div>
    </div>
    <keep-alive-stub
      include="MeetupsCalendar"
    >
      <meetups-list-stub
        meetups="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
      />
    </keep-alive-stub>
  </ui-container-stub>
</div>
`;
